src/main/java/school/lemon/changerequest/java/container/ContainerImpl.java

```diff
@@ -55,49 +61,28 @@ public boolean add(int element, int index) {
 
     @Override
     public boolean remove(int index) {
-        for (int i = 0; i < size; i++)
-            if (index == i) {
-                fastRemove(i);
-                return true;
-            }
+        if (index < size && index >= 0) {
+            int numMoved = size - index - 1;
+            if (numMoved > 0)
+                System.arraycopy(elementData, index + 1, elementData, index,
+                        numMoved);
+            elementData[--size] = 0;
+            return true;
+        }
         return false;
     }
 
-    private void fastRemove(int index) {
-        int numMoved = size - index - 1;
-        if (numMoved > 0)
-            System.arraycopy(elementData, index + 1, elementData, index,
-                    numMoved);
-        elementData[--size] = 0;
-    }
-
     private void ensureCapacityInternal(int minCapacity) {
-        if (elementData.length == INITIAL_ARRAY_SIZE) {
-            minCapacity = Math.max(INITIAL_ARRAY_SIZE, minCapacity);
+        if (minCapacity >= INITIAL_ARRAY_SIZE) {
+            if (minCapacity - MAX_ARRAY_SIZE > 0) {
+                minCapacity = hugeCapacity(minCapacity);
+            }
+            elementData = Arrays.copyOf(elementData, minCapacity);
```

Comment:
For sure it's not a good idea to increase array size by 1 each time new element is added.
Please use some multiplier.